### PR TITLE
Add 'successes' to result of data test validator

### LINKED ISSUE #397: Return all data tests run

### PR DESCRIPTION
**Pull request by DylanBaker**
## Change description

We have had a request from a number of customers to return _all_ data tests, not just passing data tests. This PR achieves this by doing the following:

- Adds a `successes` key to the `Explore` class.
- Adds all passing data tests as JSON to the `successes` key on the explore.
- Formats the results for consumption by the hosted service

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

Closes #397 

### Security

- [x] Security impact of change has been considered
- [x] Code follows security best practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer

### CODE REVIEW

File: spectacles/validators/data_test.py [hunk 1]
```diff
@@ -50,6 +50,16 @@ def validate(self) -> Dict[str, Any]:
             model_name = test["model_name"]
             explore_name = test["explore_name"]
             query_url_params = test["query_url_params"]
+            file_path = test["file"].split("/", 1)[1]
```
Comment (joshtemple) [inline spectacles/validators/data_test.py:53]: Is this reliable for all paths? Worried about KeyErrors happening here.

File: spectacles/validators/data_test.py [hunk 2]
```diff
@@ -50,6 +50,16 @@ def validate(self) -> Dict[str, Any]:
...
+            explore_url = (
+                f"{self.client.base_url}/explore/{model_name}"
+                f"/{explore_name}?{query_url_params}"
+            )
```
Comment (joshtemple) [inline spectacles/validators/data_test.py:57]: I guess the earlier PR didn't actually implement this?

File: spectacles/validators/data_test.py [hunk 3]
```diff
@@ -50,6 +50,16 @@ def validate(self) -> Dict[str, Any]:
...
+            lookml_url = (
+                f"{self.client.base_url}/projects/{self.project.name}"
+                f"/files/{file_path}?line={test['line']}"
+            )
```
Comment (joshtemple) [inline spectacles/validators/data_test.py:61]: Nice! Hadn't realized we had everything needed to return this.

**Review by joshtemple — COMMENTED** [review]
Generally looks good. I would like to see a test or two before we merge. Noting for viewers that we're planning to refactor this work a bit more cleanly in a future PR, this is a short-term implementation.

File: spectacles/validators/data_test.py (hunk 2 above)
Comment (DylanBaker) [reply spectacles/validators/data_test.py:57]: It did. I'm just moving it outside of the loop below as we already have all the info and need it for both the error and success path so it stops us from declaring it twice.

File: spectacles/validators/data_test.py (hunk 3 above)
Comment (DylanBaker) [reply spectacles/validators/data_test.py:61]: I think you had actually done the first version of this. I'm just pulling it up from the later loop.

File: spectacles/validators/data_test.py (hunk 1 above)
Comment (DylanBaker) [reply spectacles/validators/data_test.py:53]: Which bit are you worried about? I'm pretty confident the `file` key is on every test returned and it is always prefixed with `projectname/`.

**Comment by DylanBaker** [comment]
@joshtemple I'll definitely add some more tests. Otherwise, questions answered in thread.

File: spectacles/validators/data_test.py (hunk 1 above)
Comment (joshtemple) [reply spectacles/validators/data_test.py:53]: A completely unfounded concern that the path format might differ in different setups. I would be satisfied if we catch an `IndexError` when we do the `split` + indexing and return a sensible error message about the path having an unexpected format.

**Review by joshtemple — APPROVED** [review]

**Pull request merged**